### DISCUSSION TITLE
How to make a button and a mouse in cosmos?

### DISCUSSION THREAD

**Discussion by emanueleriboldi** [CosmosOS/Cosmos#1631] [Q&A]
**How to make a button and a mouse in cosmos?**

**Comment by quajak** [comment]
Do you already have the start of a GUI? Cosmos gives you access to the mouse position, to draw that on screen is up to you. See this for some more info https://github.com/CosmosOS/Cosmos/wiki/Mouse For buttons, its similar, Cosmos does not have out of the box support for such high level GUI features. Your best bet is to either use someone elses GUI framework or start developing your own. Implementing a button can be as easy as drawing a square and checking if a mouse button is pressed.

**Comment by emanueleriboldi** [comment]
but the build mouse link is not working!

**Reply by quajak** [reply]
True, I removed that part. The process is still the same, use what is stated there to get the data from the mouse. How you choose to display it/work with it, is up to you

**Comment by Andry-lab** [comment]
how i create button like ms-dos??

**Reply by FreeBSoD** [reply]
> how i create button like ms-dos??

I am also making a DOS like os, I also want to know.

**Comment by Andry-lab** [comment]
And, so how i create buttons??

**Reply by quajak** [reply]
The simplest button would be to draw a rectangle and then check if the mouse if over the button when the user clicks.

**Comment by Andry-lab** [comment]
How?

Inviato da iPhone
